### PR TITLE
Add rigetti m3 as a new target

### PR DESCRIPTION
**Pull request by shenyingjun**

### CODE REVIEW

File: azure-quantum/azure/quantum/target/rigetti/target.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class RigettiTarget(str, Enum):
 
     ASPEN_11 = "rigetti.qpu.aspen-11"
     ASPEN_M_2 = "rigetti.qpu.aspen-m-2"
+    ASPEN_M_3 = "rigetti.qpu.aspen-m-3"
```
Comment (shenyingjun) [inline azure-quantum/azure/quantum/target/rigetti/target.py:34]: Need to confirm the name

**Comment by jselig-rigetti** [comment]
Sorry, I didn't see this before I made #404 - do you want me to close that one?

**Comment by shenyingjun** [comment]
> 
It's fine, feel free to close the other PR. thanks!

**Comment by jselig-rigetti** [comment]
I've closed #404 as a duplicate of this @ashwinmayya @shenyingjun

**Review by masenol — APPROVED** [review]

**Pull request merged**